kernel/src/table_changes/mod.rs

```diff
@@ -92,19 +98,18 @@ impl TableChanges {
             Snapshot::try_new(table_root.as_url().clone(), engine, Some(start_version))?;
         let end_snapshot = Snapshot::try_new(table_root.as_url().clone(), engine, end_version)?;
 
-        // Verify CDF is enabled at the beginning and end of the interval to fail early. We must
-        // still check that CDF is enabled for every metadata action in the CDF range.
-        let is_cdf_enabled = |snapshot: &Snapshot| {
-            snapshot
-                .table_properties()
-                .enable_change_data_feed
-                .unwrap_or(false)
-        };
-        if !is_cdf_enabled(&start_snapshot) {
-            return Err(Error::change_data_feed_unsupported(start_version));
-        } else if !is_cdf_enabled(&end_snapshot) {
-            return Err(Error::change_data_feed_unsupported(end_snapshot.version()));
-        }
+        // Verify CDF is enabled at the beginning and end of the interval using
+        // [`check_cdf_table_properties`] to fail early. This also ensures that column mapping is
+        // disabled.
+        //
+        // We also check the [`Protocol`] using [`ensure_cdf_read_supported`] to verify that
+        // we support CDF with those features enabled.
+        //
+        // Note: We must still check that each metadata and protocol action in the CDF range.
+        ensure_cdf_read_supported(start_snapshot.protocol())?;
+        check_cdf_table_properties(start_snapshot.table_properties(), start_snapshot.version())?;
+        ensure_cdf_read_supported(end_snapshot.protocol())?;
+        check_cdf_table_properties(end_snapshot.table_properties(), end_snapshot.version())?;
```

Comment:
combine into one and don't need version - we can just map the error returned to tack on the error here? WDYT? (probably Error::unsupported)

Reply:
As discussed, we need to process protocol and metadata separately for now since the CDF log replay may only encounter a metadata in a commit. Or it may only encounter a protocol in the commit.